### ISSUE TITLE
In certain setups found services's A-records are not resolved

### ISSUE THREAD

**Issue by mirko** [cbrand/micropython-mdns#29]
I'm experiencing that in certain setups (potentially related to chained / double-NAT setups), services are found - but not resolved to IP addresses.

What's interesting, though is, that I'm using `TXTServiceDiscovery` and the TXT records are resolved reliably and correctly, so it's just the A(?)-records which are failing.

Example output for `query_once()`: `(<ServiceResponse name=shellyem3-AAAAAAAAAAAA._http._tcp.local priority=0 weight=0 target=shellyem3-AAAAAAAAAAAA.local port=80 ips=set() txt={'discoverable': ['false'], 'arch': ['esp8266'], 'app': ['shellyem3'], 'fw_version': ['1.0'], 'id': ['shellyem3-AAAAAAAAAAAA'], 'fw_id': ['20230913-114244/v1.14.0-gcb84623']}>, <ServiceResponse name=shellyem3-BBBBBBBBBBBB._http._tcp.local priority=0 weight=0 target=shellyem3-BBBBBBBBBBBB.local port=80 ips=set() txt={'discoverable': ['true'], 'arch': ['esp8266'], 'app': ['shellyem3'], 'fw_version': ['1.0'], 'id': ['shellyem3-BBBBBBBBBBBB'], 'fw_id': ['20230913-114244/v1.14.0-gcb84623']}>)`

When I add a linux running appliance to the same network and e.g. run `avahi -a -r` results are shown correctly:
```
root@OpenWrt:~# avahi-browse -a -r
+  wlan0 IPv4 shellyem3-BBBBBBBBBBBB                        _http._tcp           local
+  wlan0 IPv4 shellyem3-AAAAAAAAAAAA                        _http._tcp           local
=  wlan0 IPv4 shellyem3-BBBBBBBBBBBB                        _http._tcp           local
   hostname = [shellyem3-BBBBBBBBBBBB.local]
   address = [192.168.188.37]
   port = [80]
   txt = ["discoverable=true" "fw_id=20230913-114244/v1.14.0-gcb84623" "fw_version=1.0" "app=shellyem3" "arch=esp8266" "id=shellyem3-244CAB436117"]
=  wlan0 IPv4 shellyem3-AAAAAAAAAAAA                        _http._tcp           local
   hostname = [shellyem3-AAAAAAAAAAAA.local]
   address = [192.168.188.38]
   port = [80]
   txt = ["discoverable=false" "fw_id=20230913-114244/v1.14.0-gcb84623" "fw_version=1.0" "app=shellyem3" "arch=esp8266" "id=shellyem3-AAAAAAAAAAAA"]

```

**Comment by mirko** [comment]
After some more debugging, it appears we're bailing out early out of the `_on_a_record()`-method - because this condition is true: https://github.com/cbrand/micropython-mdns/blob/main/src/mdns_client/service_discovery/discovery.py#L251

**Comment by mirko** [comment]
With the following patch applied:
```
src/mdns_client/service_discovery/discovery.py
@@ .. @@ class ServiceDiscovery: 
         if record.name not in self._records_by_target:
+            print(" ---> ", record.name, "not in records_by_target", self._records_by_target)
             return
```
I get the following output:
```
[..]
MDNS Discovery: Got SRV message for existing service shellyem3-AAAAAAAAAAAA._http._tcp.local
MDNS Discovery: Got SRV message for existing service shellyem3-BBBBBBBBBBBB._http._tcp.local
 --->  shellyem3-AAAAAAAAAAAA.local not in records_by_target {'shellyem3-AAAAAAAAAAAA._http._tcp.local': {<ServiceResponse name=shellyem3-AAAAAAAAAAAA._http._tcp.local priority=0 weight=0 target=shellyem3-244cab436117.local port=80 ips=set()>}, 'shellyem3-BBBBBBBBBBBB._http._tcp.local': {<ServiceResponse name=shellyem3-BBBBBBBBBBBB._http._tcp.local priority=0 weight=0 target=shellyem3-485519d6bcd6.local port=80 ips=set()>}, 'shellyem3-485519d6bcd6.local': {<ServiceResponse name=shellyem3-BBBBBBBBBBBB._http._tcp.local priority=0 weight=0 target=shellyem3-485519d6bcd6.local port=80 ips=set()>}, 'shellyem3-244cab436117.local': {<ServiceResponse name=shellyem3-AAAAAAAAAAAA._http._tcp.local priority=0 weight=0 target=shellyem3-244cab436117.local port=80 ips=set()>}}
MDNS Discovery: Got SRV message for existing service shellyem3-BBBBBBBBBBBB._http._tcp.local
```

**Comment by cbrand** [comment]
Thanks for the debugging! That helped getting to the root cause much faster. And it is an interesting one.

The library currently discards responses which have not been seen as "requested" in an SRV record to conserve memory instead of buffering all MDNS records.

It seems like the shellys, however, first send the A record and afterwards it SRV record which results into the A record being first processed and discarded as it is not seen as relevant to the library.

I guess I need to use some more ESP32 memory and add a small buffer to keep the A records stored even if they are not relevant to support devices first sending the A record before sending the SRV response.

**Comment by cbrand** [comment]
Fixed an tested with https://github.com/cbrand/micropython-mdns/commit/078f3df5cb44b4438049a26431ae09aea48b7624

Will be made available in Version 1.5.2

**Comment by mirko** [comment]
Looking good, thank you very much!
